coin/coin.go

```diff
@@ -26,16 +26,20 @@ const (
 )
 
 // NewCoin creates a new coin object
-func NewCoin(whole int64, fractional int64,
-	ticker string) Coin {
-
+func NewCoin(whole int64, fractional int64, ticker string) Coin {
 	return Coin{
 		Whole:      whole,
 		Fractional: fractional,
 		Ticker:     ticker,
 	}
 }
 
+// NewCoinp returns a pointer to a new coin.
+func NewCoinp(whole, fractional int64, ticker string) *Coin {
```

Comment:
pp: I don't see big value in adding another constructor with pointer result for the tests. I would suggest you make `coin` a non null type in the protobuf via annotations so that you don't have to deal with this case.

Reply:
There are multiple places where [`coinp` is declared](https://github.com/iov-one/weave/search?q=coinp&unscoped_q=coinp). This is the fourth time I was declaring it so I thought it deserves its own function.

Using non null coin reference is problematic because I want to compare with a coin pointer. This is the case in many places where a result is a coin pointer and that is what you want to check becuase a result can be both a zero value coin instance and `nil`.

Does this make sense?

Reply:
I have seen (and used) this work-around in lots of test code. Fair enough to bring it here as a helper.